### PR TITLE
Wave A — architecture-review hygiene (#224 #227 #228 #239 #240 #241 #242 #244 #245)

### LINKED ISSUE #224: Interpolation alpha ignores the speed multiplier — motion snaps every tick at 2×/4×
## Bug

Render interpolation divides the accumulator by the **fixed** `MS_PER_TICK` constant, but the game loop runs ticks at `MS_PER_TICK / speedMultiplier`. At 2× the alpha only reaches ~0.5 before the tick fires and `prevState` is replaced, so every entity visually jumps the remaining ~50% of its per-tick displacement at each tick boundary (~75% at 4×). At 1× behavior is correct.

## Evidence

- `src/render/game-scene.ts:1417` — `getMsPerTick: () => MS_PER_TICK / this.speedMultiplier` (the loop's real tick period)
- `src/render/game-scene.ts:1991` — `const alpha = this.gameLoop.accumulatorMs / MS_PER_TICK` (interpolation ignores the multiplier)

## Fix

Divide by the loop's current tick period (same expression as `getMsPerTick()`), not the constant.

## Why now

Beyond 2×/4× smoothness: any future tick-period change (mobile battery throttling in Phase 6, tick negotiation in Phase 7 multiplayer) silently degrades motion smoothness the same way. One-line fix; render-only, no determinism exposure.

_Surfaced by the 2026-07-03 architecture review; verified against `main` @ 500287f._

### PR DESCRIPTION
**Pull request by LightAxe**
## Summary

**Wave A of the 2026-07-03 architecture review** — the "quick wins: no simVersion, self-contained" slice, shipped as one all-or-nothing PR of per-issue commits (docs last). **No `simVersion` bump and no change to game behavior** — the whole diff is byte-identical at the sim layer (proven by the byte-gate, below). The only render-visible change is #224 (interpolation smoothing at 2×/4×).

Closes #224, #227, #228, #239, #240, #241, #242, #244, #245.

| Commit | Issue | What |
|---|---|---|
| `fix(render)` | **#224** | Interpolation alpha divides by the loop's live tick period (`MS_PER_TICK / speedMultiplier`), not the constant — fixes per-tick position snapping at 2×/4×. Pure `computeInterpAlpha` helper (game-scene.ts is coverage-excluded). |
| `test(sim)` | **#227** | Explicit generous inline timeouts (30s/20s) on the two long statistical tests so `npm run test:coverage` exits 0 under v8 instrumentation. No global timeout, no #188 CI-posture change. |
| `test,build` | **#241** | Contract tests for the three guard scripts (fixture-driven); ban dynamic `import()` in `src/sim` (with a last-wins test-file override that re-applies the float/division bans); `check-foraging-survival.ts` asserts its thresholds and exits non-zero. |
| `ci` | **#239** | `build` + `build:lib` compile-gate in CI; race-safe bump-website push (fetch+rebase+retry, `rebase --abort` on conflict, deduped failure notifier); `gameVersion` = `${pkg.version}+${gitSHA}` from a shared `scripts/version.mjs`. |
| `test(e2e)` | **#240** | Canvas-click geometry sourced from one place (`tests/helpers/geometry.ts`, evaluating the pure layout modules); ~8 inline `CANVAS_W=800` derivations + raw-literal clicks removed across three specs; `check-e2e-geometry.sh` guard (in `verify`) reddens on a re-introduced literal in either click idiom. |
| `refactor(sim)` | **#242** | Skip the provably-all-zero underground pheromone grids in the per-tick decay sweep (behavior-identical → no simVersion; grids stay allocated for save-shape stability); delete dead exports `depositDanger`/`sampleGradient`; centralize the key-format contract. |
| `feat(save)` | **#228** | Honor the acceptance window: `MIN_ACCEPTED_SIM_VERSION` stays put while LATEST advances; a version-scoped `DELIBERATE_WINDOW_BREAK_AT` escape hatch + `version-policy.test.ts` guard make raising MIN a deliberate, test-visible exception (it wipes real players' saves). No gate reaped — that's a tracked follow-up. |
| `docs` | **#244 / #245** | Claim-by-claim doc sweep (README / AGENTS / CONTRIBUTING / ARCHITECTURE / templates) written last so it reflects this branch's reality; new PR template; `Math.atan2` added to the sim lint. #245: branch-protection docs now match the platform (required approving-review count is deliberately 0 on this solo repo; review enforced by process). |

Two small fixups ride along: a prettier wrap of #224's call (the vitest-only check missed `format:check`) and the bump-website notifier dedup (internal ship-review advisory).

## Test plan

- [x] `npm run verify` — **exit 0**, 2727 tests, all four guards clean (sim-boundary, asset-path, e2e-geometry, ant-cycle)
- [x] `npm run test:coverage` — **exit 0**, 2727 passed (the two statistical tests no longer time out under instrumentation)
- [x] `npm run build` + `npm run build:lib` — both succeed; baked `gameVersion` = `0.0.1+<sha>`
- [x] `npm run test:e2e` — **26 passed**; geometry guard hand-revert proven red per-spec
- [x] **byte-gate** capture (main) → verify (this branch) — **byte-identical**: no wave-A change alters sim `WorldState`
- [x] `check-foraging-survival.ts` — exit 0 (200 seeds, all thresholds PASS)
- [x] Internal `ship-review` — `passed: true`, 0 actionable findings

**UAT:** #224 — set speed to 2×/4× in the HUD and confirm ant motion is smooth (no per-tick position snapping). Everything else is command-/CI-verified.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added build version identification to the app via a centralized version source.
* **Bug Fixes**
  * Improved render interpolation to clamp edge cases for smoother, more stable visuals.
  * Adjusted save/version compatibility policy and strengthened underground pheromone grid preservation/skip behavior.
* **Tests**
  * Expanded Playwright/Vitest coverage for geometry, sim version policy, interpolation, and pheromone key/decay rules.
* **Documentation**
  * Updated local setup, useful scripts, and bug-reporting instructions for debugging workflow.

### CODE REVIEW

**Comment by LightAxe** [comment]
@codex review

File: src/render/boot-overlay-layout.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+// src/render/boot-overlay-layout.ts
+// Phaser-free canvas-local rects for the boot-time overlays (SavePrompt,
+// GameOver, DifficultySelect). Extracted from ui-scene.ts (#240) so Playwright
+// specs — which run in Node and crash if they import ui-scene.ts (it transitively
+// imports Phaser, which touches `window` at module load) — can import these
+// coordinates from ONE source of truth instead of hand-duplicating them.
+// ui-scene.ts re-exports them to preserve its public surface; values unchanged.
+
+/** Canvas-local rect for the SavePrompt "Continue" button. */
+export const SAVE_PROMPT_CONTINUE_RECT = { x: 300, y: 280, w: 120, h: 32 } as const;
```
[the remark this answers is not in the document]
Comment (LightAxe) [reply src/render/boot-overlay-layout.ts:20]: Thanks — this is a real observation, but scoping it to #238 (owner call). These six rects were already module-scope constants in `ui-scene.ts` on `main`; #240 relocated them **verbatim** to make them Phaser-free-importable by the specs (its invariant is zero coordinate change). Converting them to `LayoutContext` functions is #238's tracked scope ("residual fixed-canvas geometry") — and the save-prompt rects aren't cleanly centered (x=300 → center 360, not canvas-center 400), so the reflow anchors are a deliberate #238 design decision rather than a mechanical rewrite. Documented the residual in `boot-overlay-layout.ts` and added this module to #238.

**Comment by LightAxe** [comment]
**Review dispositions (resolving threads):**

- **Codex P1 / CodeRabbit — boot-overlay-layout.ts LayoutContext**: deferred to #238 (owner call). These six rects were relocated verbatim from ui-scene.ts (`main`), #240's invariant is zero coordinate change, and #238 owns the `LayoutContext` conversion ("residual fixed-canvas geometry"). Documented in-file + added to #238.
- **CodeRabbit — AGENTS.md verify summary (add build/build:lib)**: not applicable. `npm run verify` (package.json) does **not** run the builds — those are separate CI *job* steps in ci.yml, run after `verify`. The AGENTS.md line describes the `verify` script and is correct; the ci.yml header comment already notes the build gate.
- **CodeRabbit — ARCHITECTURE.md fpDiv vs division ban**: already consistent. AGENTS.md:101 bans the `/` operator **and** states reviewers read inline rule-disables; `fpDiv` is exactly that single sanctioned inline-disabled exception, which ARCHITECTURE.md now documents. No contradiction.
